metrics/firebase/functions/test/main_test.dart

```diff
@@ -450,6 +454,441 @@ void main() {
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(DocumentData.fromMap(newBuildJson));
+        whenCreateTaskDocument().thenAnswer((_) => Future.value());
+
+        await onBuildUpdatedHandler(change, null);
+
+        final dataMatcher = predicate<DocumentData>((data) {
+          oldBuildJson['startedAt'] = oldBuildJson['startedAt'].toDateTime();
+          newBuildJson['startedAt'] = newBuildJson['startedAt'].toDateTime();
+          final expectedData = {
+            'oldBuild': oldBuildJson,
+            'newBuild': newBuildJson,
+          };
+
+          return const DeepCollectionEquality().equals(
+            data.getNestedData('data').toMap(),
+            expectedData,
+          );
+        });
+
+        verify(collectionReference.add(argThat(dataMatcher))).called(1);
+      },
+    );
+
+    test(
+      "creates a task document with context equals to the error string representation if updating the build day's document data fails",
+      () async {
+        final exception = Exception('test');
+
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(buildDocumentData);
+        whenCreateTaskDocument(exception: exception)
+            .thenAnswer((_) => Future.value());
+
+        await onBuildUpdatedHandler(change, null);
+
+        final contextMatcher = predicate<DocumentData>(
+          (data) => data.getString('context') == exception.toString(),
+        );
+
+        verify(collectionReference.add(argThat(contextMatcher))).called(1);
+      },
+    );
+
+    test(
+      "creates a task document with createdAt equals to the current date time if updating the build day's document data fails",
+      () async {
+        final currentDateTime = DateTime.now();
+        final expectedCreatedAt = Timestamp.fromDateTime(currentDateTime);
+
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(buildDocumentData);
+        whenCreateTaskDocument().thenAnswer((_) => Future.value());
+
+        await withClock(Clock.fixed(currentDateTime), () async {
+          await onBuildUpdatedHandler(change, null);
+
+          final createdAtMatcher = predicate<DocumentData>((data) {
+            return data.getTimestamp('createdAt') == expectedCreatedAt;
+          });
+
+          verify(collectionReference.add(argThat(createdAtMatcher))).called(1);
+        });
+      },
+    );
+
+    test(
+      "deletes an existing task if the build day data updates successfully",
+      () async {
+        whenAccessChangeData(documentSnapshot: newDocumentSnapshot)
+            .thenReturn(buildDocumentData);
+        whenTaskDocuments().thenReturn([documentSnapshot]);
+
+        await onBuildUpdatedHandler(change, null);
+
+        verify(documentReference.delete()).called(1);
+      },
+    );
+
+    test(
+      "does not delete an existing task if updating the build day's document data fails",
```

Comment:
Should we test that this function does not delete any task if the task does not exist?